module/VuFind/src/VuFind/AjaxHandler/GetResultCount.php

```diff
@@ -0,0 +1,105 @@
+<?php
+/**
+ * Ajax Controller for Libraries Extension
+ *
+ * PHP version 5
+ *
+ * Copyright (C) Staats- und Universitätsbibliothek 2017.
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 2,
+ * as published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
+ *
+ * @category VuFind2
+ * @package  Controller
+ * @author   Hajo Seng <hajo.seng@sub.uni-hamburg.de>
+ * @license  http://opensource.org/licenses/gpl-2.0.php GNU General Public License
+ * @link     https://github.com/subhh/beluga
+ */
+namespace VuFind\AjaxHandler;
+
+use Laminas\Mvc\Controller\Plugin\Params;
+use Laminas\Stdlib\Parameters;
+use VuFind\Search\Results\PluginManager as ResultsManager;
+
+/**
+ * This controller handles global AJAX functionality
+ *
+ * @category VuFind2
+ * @package  Controller
+ * @author   Chris Hallberg <challber@villanova.edu>
+ * @license  http://opensource.org/licenses/gpl-2.0.php GNU General Public License
+ * @link     http://vufind.org/wiki/vufind2:building_a_controller Wiki
+ */
+class GetResultCount extends AbstractBase
+{
+    /**
+     * ResultsManager
+     *
+     * @var resultsManager
+     */
+    protected $resultsManager;
+
+    /**
+     * Constructor
+     *
+     * @param ResultsManager $resultsManager Results Manager
+     */
+    public function __construct(ResultsManager $resultsManager)
+    {
+        $this->resultsManager = $resultsManager;
+    }
+
+    /**
+     * Handle a request.
+     *
+     * @param Params $params Parameter helper from controller
+     *
+     * @return array [response data, HTTP status code]
+     */
+    public function handleRequest(Params $params)
+    {
+        $queryString = $params->fromQuery('querystring');
+        $queryString = urldecode(
+            str_replace(
+                '&amp;',
+                '&',
+                trim($queryString)
+            )
+        );
+
+        $queryArray = explode('&', $queryString);
```

Comment:
Did you consider using PHP's built-in `parse_url` function? I imagine that might simplify some of this code, unless there's a specific reason why it won't work.

Reply:
I'm going to try this as well; I'm not sure if I've done that so far.

Reply:
I did some quick experimentation -- I was able to use `parse_url` to extract query strings from URLs, and then `parse_str` to convert the resulting query strings into PHP arrays. It seemed to handle brackets correctly.

Reply:
It's a bit weird that `parse_str` sets the PHP array by reference instead of returning a value, but that was the biggest confusion I ran into.